### PR TITLE
SLING-10967 add support to register Gauges directly without whiteboar…

### PR DESCRIPTION
**Pull request by joerghoh**
…d support

### CODE REVIEW

File: src/main/java/org/apache/sling/commons/metrics/Gauge.java [hunk 1]
```diff
@@ -34,14 +43,12 @@
...
- * <p> A Gauge instance should be registered with OSGi ServiceRegistry with {@code Gauge#NAME} set
- * to Gauge name. Then the Gauge instance would be registered with MetricService via whiteboard
- * pattern
+
  *
  * @param <T> the type of the metric's value
  */
 @ConsumerType
-public interface Gauge<T> {
+public interface Gauge<T> extends Metric {
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/commons/metrics/Gauge.java:51]: I fear we need to create a new interface - Gauge is right now a ConsumerType - this new gauge should be a ProviderType. As Metric is a ProviderType Gauge can't extend Metric.

Comment (joerghoh) [reply src/main/java/org/apache/sling/commons/metrics/Gauge.java:51]: So is the problem the fact that the "new Gauge" should be a ProviderType and this one is a ConsumerType? Isn't it possible to use this interface for both cases?

Or is the problem rather that this ConsumerType extends a Providertype?

Comment (joerghoh) [reply src/main/java/org/apache/sling/commons/metrics/Gauge.java:51]: ok, I understand.

Extending the ```Metric``` interface is also not really necessary from an API point of view, it is only needed at the implementation level. I will change the implementation so the existing Gauge interface (as a ConsumerType) will stay as it currently is.

**Review by cziegeler — APPROVED** [review]
Lgtm, thanks!

**Pull request merged**